addons/mail/controllers/discuss_controller.py

```diff
@@ -0,0 +1,284 @@
+    def fetch_profile_picture(self, channel_id, partner_id, **kwargs):
+        channel_partner = self._get_channel_partner(channel_id)
+        target_channel_partner = channel_partner.env['mail.channel.partner'].search([
+            ('channel_id', '=', channel_id),
+            ('partner_id', '=', partner_id),
+        ], limit=1)
+        
+        # Ensure both channel partners exist and are members of the channel.
+        if not (channel_partner and target_channel_partner):
+            raise NotFound()
+        
+        return channel_partner.env['ir.http'].content_image(
+            field='avatar_128',
+            id=partner_id,
+            model='res.partner',
+        )
+    
+    @http.route('/mail/channel/<int:channel_id>/image/<int:attachment_id>/<int:width>x<int:height>', type='http', auth='public')
+    def fetch_image(self, channel_id, attachment_id, width, height, **kwargs):
+        # Ensure the current user or guest is member of the channel.
+        channel_partner = self._get_channel_partner(channel_id)
+        if not channel_partner:
+            raise NotFound()
+
+        # Ensure the attachment is available in the channel.
+        if not channel_partner.env['ir.attachment'].search([
+            ('id', '=', attachment_id),
+            ('res_id', '=', channel_id),
+            ('res_model', '=', 'mail.channel'),
+        ], limit=1):
+            raise NotFound()
+        
+        return channel_partner.env['ir.http'].content_image(
+            id=attachment_id,
+            height=height,
+            width=width,
+        )
+
+    @http.route('/mail/channel/<int:channel_id>/content/<int:attachment_id>', type='http', auth='public')
+    def download_content(self, channel_id, attachment_id, **kwargs):
+        # Ensure the current user or guest is member of the channel.
+        channel_partner = self._get_channel_partner(channel_id)
+        if not channel_partner:
+            raise NotFound()
+
+        # Ensure the attachment is available in the channel.
+        if not channel_partner.env['ir.attachment'].search([
+            ('id', '=', attachment_id),
+            ('res_id', '=', channel_id),
+            ('res_model', '=', 'mail.channel'),
+        ], limit=1):
+            raise NotFound()
+
+        return channel_partner.env['ir.http']._content_common(
+            download=True,
+            field='datas',
+            id=attachment_id,
+            model='ir.attachment',
+        )
+
+    def _fetch_messages(self, domain, min_id, max_id, limit=20):
+        if max_id:
+            domain.append(('id', '>', max_id))
+        if min_id:
+            domain.append(('id', '<', min_id))
+        return request.env['mail.message'].sudo().message_fetch(domain, limit)
+
+    def _get_channel(self, channel_id):
+        return request.env['mail.channel'].sudo().search([('id', '=', channel_id)], limit=1)
```

Comment:
why not use browse().exists()?

Reply:
No specific reason, this could be changed as suggested.